core/dbt/events/base_types.py

```diff
@@ -3,18 +3,62 @@
 import threading
 from datetime import datetime
 
+import functools
+import collections
 
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 # These base types define the _required structure_ for the concrete event #
 # types defined in types.py                                               #
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
 
+# TODO: this should not live here - duplicated and modified from utils.py because of circular imports for now
```

Comment:
This is here temporarily to test out if I could get it working.  Importing the one in `util.py` causes some circular dependency issues.